src/core/featuremodel.cpp

```diff
@@ -229,26 +317,53 @@ bool FeatureModel::save()
     rv = false;
   }
 
-  QgsFeature feat = mFeature;
-  if ( !mLayer->updateFeature( feat ) )
-    QgsMessageLog::logMessage( tr( "Cannot update feature" ), QStringLiteral( "QField" ), Qgis::Warning );
-
-  if ( QgsProject::instance()->topologicalEditing() )
+  switch ( mModelMode )
   {
-    applyVertexModelToLayerTopography();
-    mLayer->addTopologicalPoints( feat.geometry() );
-  }
+    case SingleFeatureModel:
+    {
+      QgsFeature feat = mFeature;
+      if ( !mLayer->updateFeature( feat ) )
+        QgsMessageLog::logMessage( tr( "Cannot update feature" ), QStringLiteral( "QField" ), Qgis::Warning );
+
+      if ( QgsProject::instance()->topologicalEditing() )
+      {
+        applyVertexModelToLayerTopography();
+        mLayer->addTopologicalPoints( feat.geometry() );
+      }
 
-  rv &= commit();
+      rv &= commit();
```

Comment:
I don't see any issue in this part of the code. But on testing I receive back a message in the log `Could not save changes. Rolling back.` Although it apparently stores the changes. This happens on editing a single feature. With multi editing there is no message. Maybe you can reproduce. Otherwise I can check it...

Reply:
@signedav , fixed.